that_depends/providers/attr_getter.py

```diff
@@ -13,14 +13,20 @@ def _get_value_from_object_by_dotted_path(obj: typing.Any, path: str) -> typing.
     return attribute_getter(obj)
 
 
-class AttrGetter(AbstractProvider[T]):
+class AttrGetter(
+    AbstractProvider[T],
+):
     __slots__ = "_provider", "_attrs"
 
     def __init__(self, provider: AbstractProvider[T], attr_name: str) -> None:
         self._provider = provider
         self._attrs = [attr_name]
 
     def __getattr__(self, attr: str) -> "AttrGetter[T]":
+        if attr.startswith("_"):
+            msg = f"'{type(self)}' object has no attribute '{attr}'.\
+                Late resolution of attributes starting with `_` is not supported."
+            raise AttributeError(msg)
```

Comment:
better to stick to default error message as in https://github.com/modern-python/that-depends/blob/main/that_depends/providers/singleton.py#L26